frontend/src/app/main/data/workspace/groups.cljs

```diff
@@ -149,17 +162,22 @@
   (ptk/reify ::ungroup-selected
     ptk/WatchEvent
     (watch [it state _]
-      (let [page-id   (:current-page-id state)
-            objects   (wsh/lookup-page-objects state page-id)
-            is-group? #(or (= :bool (:type %)) (= :group (:type %)))
-            lookup    #(get objects %)
-            prepare   #(prepare-remove-group it page-id % objects)
+      (let [page-id       (:current-page-id state)
+            objects       (wsh/lookup-page-objects state page-id)
+
+            prepare
+            (fn [shape-id]
+              (let [shape (get objects shape-id)]
+                (cond
+                  (or (cph/group-shape? shape) (cph/bool-shape? shape))
+                  [(remove-group-changes it page-id shape objects)]
+
+                  (cph/frame-shape? shape)
+                  [(remove-frame-changes it page-id shape objects)])))
```

Comment:
This function returns always a vector of single element, and then... `mapcat`. I think it can return the value as-is and then, use `map` instead `mapcat`

Reply:
Not exactly, there is the omitted default `nil` when it won't match any conditions. When it's nil the mapcat will not modify the resulting list.

Reply:
yep, in this case instead of `map` you can use `keep` :P

Reply:
I'll remove the mapcat for a keep